src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -283,7 +283,7 @@ export class HubConnection {
         }
     }
 
-    private _stopInternal(error?: Error): Promise<void> {
+    private async _stopInternal(error?: Error): Promise<void> {
```

Comment:
Making `_stopInternal` async seems super sketchy given how stop() is implemented:

```ts
    public async stop(): Promise<void> {
        // Capture the start promise before the connection might be restarted in an onclose callback.
        const startPromise = this._startPromise;
        this._stopPromise = this._stopInternal();
        await this._stopPromise;
```

And even if it weren't for that, aren't we now returning nested promises every time we return early? We should not try to reason about what happens when the `_stopPromise` is assigned while the `async _stopInternal` function is still running. It's probably simpler/better to fire and forget sending the close message like we did in C# and Java.

Either that, or wrap the actual async bits of this into a new `async _sendCloseMessageAndStop()` method and call that instead of calling `this.connection.stop(error);` directly.